From 09c732dc5e9684efdbd6742610b083ee3bc64bc3 Mon Sep 17 00:00:00 2001
From: agaraix <anath.bandux.garai@intel.com>
Date: Fri, 29 Nov 2019 15:59:25 +0530
Subject: [PATCH] UGW_SW-45104 : EoGRE/IPoGRE US TTL fix

TTL to be inherited from inner IP header, when not configured explicitly.
---
 include/net/ppa/ppa_stack_al.h | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/include/net/ppa/ppa_stack_al.h b/include/net/ppa/ppa_stack_al.h
index d02e256c7106..ae4a90367f4c 100644
--- a/include/net/ppa/ppa_stack_al.h
+++ b/include/net/ppa/ppa_stack_al.h
@@ -1116,6 +1116,12 @@ int ppa_get_multicast_pkt_ip(PPA_BUF *ppa_buf, void *dst_ip, void *src_ip);
 	\note
  */
 void ppa_get_pkt_dst_ip(PPA_IPADDR *ip, PPA_BUF *buf);
+/*! \brief   Returns IP TTL of the packet.
+  \param[in] ppa_buf Pointer to the packet buffer.
+  \return TTL value.
+  \note
+ */
+uint8_t ppa_get_pkt_ip_ttl(PPA_BUF *ppa_buf);
 /*! \brief Returns source TCP/UDP port of the IP packet.
 	\param[in] ppa_buf Pointer to the packet buffer.
 	\return TCP/UDP Source Port of the packet.
@@ -2253,7 +2259,7 @@ int ppa_get_hash_from_packet(PPA_BUF *ppa_buf,
 	\return ipttl
 	\note
  */
-int32_t ppa_get_ip_ttl(void);
+int32_t ppa_get_ip_ttl_default(void);
 /*! \brief get system meminfo
 	\param pointer to systeminfo structure
 	\return system mem info
